x-pack/test/functional_basic/apps/ml/permissions/full_ml_access.ts

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import path from 'path';
+
+import { FtrProviderContext } from '../../../ftr_provider_context';
+
+import { USER } from '../../../../functional/services/ml/security_common';
+
+export default function ({ getService }: FtrProviderContext) {
+  const esArchiver = getService('esArchiver');
+  const ml = getService('ml');
+
+  const testUsers = [USER.ML_POWERUSER, USER.ML_POWERUSER_SPACES];
+
+  describe('for user with full ML access', function () {
+    for (const user of testUsers) {
+      describe(`(${user})`, function () {
+        const ecIndexPattern = 'ft_module_sample_ecommerce';
+        const ecExpectedTotalCount = 287;
+        const ecExpectedFieldPanelCount = 2;
+        const ecExpectedModuleId = 'sample_data_ecommerce';
+
+        const uploadFilePath = path.join(
+          __dirname,
+          '..',
+          '..',
+          '..',
+          '..',
+          'functional',
+          'apps',
+          'ml',
+          'data_visualizer',
+          'files_to_import',
+          'artificial_server_log'
+        );
+        const expectedUploadFileTitle = 'artificial_server_log';
+
+        before(async () => {
+          await ml.api.cleanMlIndices();
+
+          await esArchiver.loadIfNeeded('ml/module_sample_ecommerce');
+          await ml.testResources.createIndexPatternIfNeeded(ecIndexPattern, 'order_date');
+
+          await ml.securityUI.loginAs(user);
+        });
+
+        after(async () => {
+          await ml.securityUI.logout();
+        });
+
+        it('should display the ML file data vis link on the Kibana home page', async () => {
+          await ml.testExecution.logTestStep('should load the Kibana home page');
+          await ml.navigation.navigateToKibanaHome();
+
+          await ml.testExecution.logTestStep('should display the ML file data vis link');
+          await ml.commonUI.assertKibanaHomeFileDataVisLinkExists();
+        });
+
+        it('should display the ML entry in Kibana app menu', async () => {
+          await ml.testExecution.logTestStep('should open the Kibana app menu');
+          await ml.navigation.openKibanaNav();
+
+          await ml.testExecution.logTestStep('should display the ML nav link');
+          await ml.navigation.assertKibanaNavMLEntryExists();
+        });
+
+        it('should display tabs in the ML app correctly', async () => {
+          await ml.testExecution.logTestStep('should load the ML app');
+          await ml.navigation.navigateToMl();
+
+          await ml.testExecution.logTestStep('should display the disabled "Overview" tab');
+          await ml.navigation.assertOverviewTabEnabled(false);
```

Comment:
question: is it worth moving `logTestStep` into `assertOverviewTabEnabled` ? Spotted it in another test

Reply:
@darnautov Good point! I already thought about if/when to move test step logging into service methods. But since the test files can arbitrarily combine service method calls, I found it hard to move them over. In the end, the test file defines, what calls to collect into one test step, e.g. to stay with this code line: `ml.navigation.assertOverviewTabEnabled` might not always be a standalone test step, but sometimes be combined with other methods to form a test step.

Reply:
I think your question points out that the test step logging in this case is too detailed. Do you think it would help to only have one test step log message for the whole tab assertion like `should display the tabs with correct enabled/disabled state`?

Reply:
`assertOverviewTabEnabled ` is just a particular example. I've noticed that quite some single assertions having `logTestStep` that could be defined in an assertion method itself. 
I believe logging is useful and allows us to identify a failed step in a more obvious way, so maybe to make test suits more concise we should move single assertion logging into the methods itself and define `logTestStep` for groups only (e.g. `should display the tabs with correct enabled/disabled state` you've mentioned)?

Reply:
Thanks for the feedback! I agree that we can improve the test step / single assertion logging. I'll think a bit more about it and will come up with a proposal in a separate PR.